File: FoodDeliveryApp/FoodDeliveryApp.API/Models/MenuItem.cs
namespace FoodDeliveryApp.API.Models;

public class MenuItem
{
    public int MenuItemId { get; set; }
    public int RestaurantId { get; set; }
    public string Name { get; set; } = null!;
    public double Price { get; set; }
    public string Description { get; set; } = null!;
    public string ImageURL { get; set; } = null!;
}

File: FoodDeliveryAppFront/FoodDeliveryAppUI/src/Features/MenuItem/MenuItem.ts
import { apiService } from "../../services/apiService";
import type { IMenuItem } from "../../Interfaces/MenuItem";

export interface CreateMenuItemRequest {
    restaurantId: number;
    name: string;
    price: number;
    description: string;
    imageURL: string;
}

export interface UpdateMenuItemRequest {
    name: string;
    price: number;
    description: string;
    imageURL: string;
}

export const createMenuItem = (data: CreateMenuItemRequest) =>
    apiService.post<CreateMenuItemRequest, IMenuItem>("/menuitem", data);

export const getMenuItemsByRestaurant = (restaurantId: number) =>
    apiService.get<IMenuItem[]>(`/menuitem/restaurant/${restaurantId}`);

export const updateMenuItem = (id: number, data: UpdateMenuItemRequest) =>
    apiService.put<UpdateMenuItemRequest, IMenuItem>(`/menuitem/${id}`, data);

export const deleteMenuItem = (id: number) =>
    apiService.delete<IMenuItem>(`/menuitem/${id}`);